05/ContactForm.js

```diff
@@ -1,10 +1,97 @@
-import React from 'react';
 
-import account from './account';
+import React, { useReducer} from 'react';
+import validateForm from './validator';
+import './styles.css';
+// import account from './account';
 
-const ContactForm = () => {
-    console.log(account);
-    return <form>??</form>;
-};
+function ContactForm() {
+    const formRef = React.createRef(null);
+    const init = {
+        name: '',
+        email: '',
+        phone: '',
+        subject: '',
+        message: '',
+        errors: null,
+    }
+    
+    function reducer (state, action){
+
+        switch(action.type){
+        case 'SET_ERRORS':
+            return {
+                ...state,
+                errors: action.payload
+            }
+        case 'SUBMIT_FORM':
+            return {
+                ...state,
+                ...action.payload,
+                success: true,
+            }
+        default: 
+            return state
+        };
+    }
+    const [state, dispatch] = useReducer(reducer, init);
+    const handleSubmit = (e, form) => {
+        e.preventDefault();
+        const{name, email, phone, subject, message} = form.current
+        const errors = validateForm(form.current);
+        const obj = {
+            name: name.value,
+            email: email.value,
+            phone: phone.value,
+            subject: subject.value,
+            message: message.value,
+            errors: null,
+    
+        }
+        if(errors === null){
+            dispatch({
+                type: 'SUBMIT_FORM',
+                payload: obj,
+            })
+            form.current.reset()
+        }else{
+            dispatch({
+                type: 'SET_ERRORS',
+                payload: errors
+            })
+        }
+    }
+
+    return (
+        <form className="form" ref={formRef} onSubmit={(e)=>handleSubmit(e, formRef)}>
+            <label className="form__label" htmlFor="name">
+                Imię i nazwisko:
+                <input className="form__input" type="text" name="name" />
+                {state.errors !== null && state.errors.name ? <span className="form__error"style={{color: "red"}}>{state.errors.name}</span> : null}
+            </label>
+            <label className="form__label" htmlFor="email">
+                E-mail:
+                <input className="form__input" type="text" name="email" />
+                {state.errors !== null && state.errors.email ? <span className="form__error" style={{color: "red"}}>{state.errors.email}</span> : null}
+            </label>
+            <label className="form__label" htmlFor="phone">
+                Telefon:
+                <input className="form__input" type="text" name="phone" />
+                {state.errors !== null && state.errors.phone ? <span className="form__error" style={{color: "red"}}>{state.errors.phone}</span> : null}
+            </label>
```

Comment:
Prawdopodobnie do wygenerowania elementów `label` (i ich zawartości) możesz wykorzystać `fields.js` i pętle -  będzie to lepiej wyglądać (mniej powielonego kodu) i łatwiej będzie wprowadzać nowe pola :)